iiod/usb-dmabuf.c

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/*
+ * libiio - Library for interfacing industrial I/O (IIO) devices
+ *
+ * Copyright (C) 2022 Analog Devices, Inc.
+ * Author: Paul Cercueil <paul.cercueil@analog.com>
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <sys/ioctl.h>
+
+#define IIO_FFS_DMABUF_ATTACH	_IOW('g', 131, int)
+#define IIO_FFS_DMABUF_DETACH	_IOW('g', 132, int)
+#define IIO_FFS_DMABUF_TRANSFER	_IOW('g', 133, struct iio_ffs_dmabuf_transfer)
+
+struct iio_ffs_dmabuf_transfer {
+	int fd;
+	uint32_t flags;
```

Comment:
maybe let's make this u64 already? I know the kernel side has u32 (which would have to be also updated) and I was actually wondering why not `unsigned long` but now I see this is used in ioctl().

BTW, the only reason I'm stating to move this into u64 is just to be on the safe side and have more space for flags (we never know how this will grow even though now, it seems that 32 flags is more than enough)

Reply:
Well I currently have a total of 0 flags, I would think that 32 available flags is more than enough :)

I could switch to u64 flags, at the cost of doubling the size of the structure (24 bytes vs. 12 bytes), but I'm not really sure that will ever be needed.

I will send a patchset for the Linux bits very soon, let's see if they say anything about that.